frontend/src/hooks/useTrip.ts

```diff
@@ -0,0 +1,51 @@
+import {useEffect, useState} from "react";
+import Trip from "../types/Trip";
+import Api from "../api/Api";
+import {isAxiosError} from "axios";
+import Location from "../types/Location";
+
+const initialState: Trip = {
+    id: "",
+    title: "",
+    tripTimestamp: "",
+    locations: []
+}
+
+export default function useTrip(id: string) {
+    const [trip, setTrip] = useState<Trip>(initialState);
+    const [notFound, setNotFound] = useState<boolean>();
+
+    useEffect(() => {
+        (async() => {
+            try {
+                const trip = await Api.getTrip(id);
+                setTrip(trip);
+            } catch (e) {
+                if (isAxiosError(e) && e.response?.status === 404) {
+                    setNotFound(true);
+                }
+            }
+        })();
+    }, [id]);
+
+    const updateTripQuery = async (trip: Trip) => {
+        const updatedTrip = await Api.updateTrip(trip);
+        setTrip(updatedTrip);
+    }
+
+    const deleteTripQuery = async (trip: Trip) => {
+        await Api.deleteTrip(trip.id);
+    };
+
+    const getShortestPathForTripQuery = async (trip: Trip) => {
+        const shortestPath = await Api.getShortestPathForTrip(trip.id);
+        setTrip({...trip, locations: shortestPath});
+    }
+
+    const removeLocationFromTrip = (deletedLocation: Location) => {
+        const updatedTrip = {...trip, locations: trip.locations.filter(location => location.id !== deletedLocation.id)}
+        setTrip(updatedTrip);
+    }
+
```

Comment:
General question: Why do we need the await command here as well as in Api.ts?

Reply:
Bei removeLocationFromTrip wird updated Trip nicht in backend geschickt?

Reply:
@ThanhXuanDuong 
ich dachte, erst wenn man Save drückt, werden alle Änderungen für den Trip gespeichert, also auch die Änderungen an den Locations. 
Weiß nicht, ob wir das wirklich so wollen? 

Was meint ihr?

Reply:
Ach ja, finde ich gut so. Alles klar!

Reply:
Ich würde auch "erst" durch den Save button alles geschlossen speichern